components/kyma-environment-broker/internal/broker/instance_create.go

```diff
@@ -427,3 +440,68 @@ func (b *ProvisionEndpoint) createDashboardURL(planID, instanceID string) string
 		return fmt.Sprintf("%s/?kubeconfigID=%s", b.dashboardConfig.LandscapeURL, instanceID)
 	}
 }
+
+func validateCidr(cidr string) (*net.IPNet, error) {
+	ip, ipNet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return nil, err
+	}
+	// find cases like: 10.250.0.1/19
+	if ipNet != nil {
+		if !ipNet.IP.Equal(ip) {
+			return nil, fmt.Errorf("%s must be valid canonical CIDR", ip)
+		}
+	}
+	return ipNet, nil
+}
+
+func (b *ProvisionEndpoint) validateNetworking(parameters internal.ProvisioningParametersDTO) error {
+	var err, e error
+	if len(parameters.Zones) > 4 {
+		// the algorithm of creating AWS zone CIDRs does not work for more than 4 zones
+		err = multierror.Append(err, fmt.Errorf("number of zones must not be greater than 4"))
+	}
+	if parameters.Networking == nil {
+		return nil
+	}
+	var nodes, services, pods *net.IPNet
+	if nodes, e = validateCidr(parameters.Networking.NodesCidr); e != nil {
+		err = multierror.Append(err, fmt.Errorf("while parsing nodes CIDR: %w", e))
+	}
+	// error is handled before, in the validate CIDR
+	cidr, _ := netip.ParsePrefix(parameters.Networking.NodesCidr)
+	if cidr.Bits() > 26 {
```

Comment:
The maximum Bits we should support is /24, because when going with /24 nodes allows assigning 3x /26 subnet for the 3 worker subnets, while there are still 8x /28 subnets, 6x of which would be used to assign to public and internal ELB subnets.

However to support this scenario, you would also need to adjust the hard-coded `workerPrefixLength := cidr.Bits() + 3` logic, as when /24 was input, you would need to go with `workerPrefixLength := cidr.Bits() + 2`, and `workerPrefixLength + 2` for `public` and `internal`

TL;DR: in general, the smallest CIDR for any workers subnet per AZ should be /26 , as it allows 62 nodes. /27 would be too low already. For Azure, we could even split /24 easily into 4x /26, but for AWS, it's trickier than the current implementation in aws_provider.go

Reply:
Or do you think if we explicitly document to the user than when providing the smallest possible /26 nodes CIDR, the runtime will only be able to scale up to maximum 6 nodes per AZ (18 nodes total) using /29 CIDR?, And furthermore on AWS they will only be able to create 2 public LoadBalancer Services (because of the /30 public CIDR per AZ)?

I think this is a big limitation though

Reply:
I'll add a validation - max value in the suffix is 23

Reply:
let's do it +2 and suffix 24